FirebaseVertexAI/Sources/Types/Internal/Imagen/ImageGenerationParameters.swift

```diff
@@ -0,0 +1,62 @@
+// Copyright 2024 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+@available(iOS 15.0, macOS 12.0, macCatalyst 15.0, tvOS 15.0, watchOS 8.0, *)
+struct ImageGenerationParameters {
+  let sampleCount: Int?
+  let storageURI: String?
+  let seed: Int32?
+  let negativePrompt: String?
+  let aspectRatio: String?
+  let safetyFilterLevel: String?
+  let personGeneration: String?
+  let outputOptions: ImageGenerationOutputOptions?
+  let addWatermark: Bool?
+  let includeResponsibleAIFilterReason: Bool?
+}
+
+// MARK: - Codable Conformance
+
+@available(iOS 15.0, macOS 12.0, macCatalyst 15.0, tvOS 15.0, watchOS 8.0, *)
+extension ImageGenerationParameters: Encodable {
+  enum CodingKeys: String, CodingKey {
+    case sampleCount
+    case storageURI = "storageUri"
+    case seed
+    case negativePrompt
+    case aspectRatio
+    case safetyFilterLevel = "safetySetting"
```

Comment:
why different names?

Reply:
IMHO `safetyFilterLevel` better expresses the meaning "Adds a filter level to safety filtering." Also planning to group `safetyFilterLevel` (`safetySetting`), `personGeneration` and `includeResponsibleAIFilterReason` (`includeRaiReason`) into a `SafetySettings` struct in the public API so this will be more distinct. That said, all the naming is still in flux.